lib/text/ui_text_displayer.js

```diff
@@ -280,7 +280,10 @@ shaka.text.UITextDisplayer = class {
       for (const element of toUproot) {
         // NOTE: Because we uproot shared region elements, too, we might hit an
         // element here that has no parent because we've already processed it.
-        if (element.parentElement) {
+        // But avoid removing regions here due to risk of missing out
+        // subsequent captions.
+        if (element.parentElement &&
+            !element.id.includes('shaka-text-region')) {
```

Comment:
This could cause regions to accumulate.  It looks like a DOM leak.  When will they actually be removed?

Reply:
I've got an alternative solution from another angle - see forthcoming PR